src/systems/filecoin_markets/deal/_index.md

```diff
@@ -3,6 +3,52 @@ menuTitle: Deals
+    - `StorageClient` and `StorageProvider` can call `WithdrawBalance` before any deal is made. (move to state X)
+- 2. `StorageClient` and `StorageProvider` negotiate a deal off chain. `StorageClient` sends a `StorageDealProposal` to a `StorageProvider`.
+    - `StorageProvider` verifies the `StorageDeal` by checking address and signature of `StorageClient`, checking the proposal has not expired, checking `StorageClient` did not call withdraw in the last X Epoch, checking both `StorageProvider` and `StorageClient` have sufficient available balances in `StorageMarketActor`.
+- 3. `StorageProvider` signs the `StorageDealProposal`  gets a `StorageDeal`.
+    - a. `StorageProvider` calls `PublishStorageDeals` in `StorageMarketActor` which will generate a `DealID` for each `StorageDeal` and store a mapping from `DealID` to `StorageDeal`. However, the deals are not active at this point.
+      - As a backup, `StorageClient` MAY call `PublishStorageDeals` with the `StorageDeal`, to activate the deal.
+      - It is possible for either `StorageProvider` or `StorageClient` to try to enter into two deals simultaneously with funds available only for one. Only the first deal to commit in the chain would clear, the second would fail with error `errorcode.InsufficientFunds`.
+    - b. `StorageProvider` calls `HandleStorageDeal` in `StorageMiningSubsystem` which will then add the `StorageDeal` into a `Sector`.
+- 4. Once the miner finishes packing a `Sector`, it generates a Sealed Sector and calls `StorageMinerActor.CommitSector` to verify the seal, store sector expiration, and record the mapping from `SectorNumber` to `SealCommitment`. It will also place this newly added `Sector` in the list of `CommittedSectors` in `StorageMinerActor`. `StorageMiner` does not earn any power for this newly added sector until its first PoSt has been submitted.
+
+# Declare and Recover Faults
+
+- 5. Declared faults are penalized to a smaller degree than spotted faults by `CronActor`. Miners declare faulty sectors by invoking `StorageMinerActor.DeclareFaults` and X of the `StorageDealCollateral` will be slashed and power corresponding to these sectors will be tempororily lost.
+- 6. Miners can then recover faults by invoking `StorageMinerActor.RecoverFaults` and have sufficient `StorageDealCollateral` in their available balances. FaultySectors are recommitted and power is only restored at the next PoSt submission.
+- 7. Sectors that are declared faulty for `storagemining.MaxFaults` consecutive ChainEpochs will result in `StoragePowerActor.SlashPledgeCollateral`.
+  - TODO: set `X` parameter
+
+# Submit PoSt
+
+(TODO: move into Storage Mining)
+
+On every PoSt Submission, the following steps happen.
+
+- 8. `StorageMinerActor` first verifies the PoSt Submission. All Sectors will be considered in `SpottedFaults` if PoSt submission has failed (move to State 14).
```

Comment:
I think this is confused:
- if an attempted PoSt submission is rejected, nothing should happen - the miner still has time to submit a correct one
- "failure" occurs when there is no submission after deadline, so you can't talk about checking it in response to the submission that didn't happen. that has to be in terms of cron.